advanced-javascript/tests/arrays.test.js

```diff
@@ -18,43 +18,108 @@ describe('Arrays', () => {
   describe('`each`', () => {
     it('should be a function', () => {
       const each = arrayFunctions.each;
-      expect(each).to.be.a('string');
+      expect(each).to.be.a('function');
     });
     // begin here
+    it('should execute call back at least once', () => {
+      const callBack = sinon.spy();
+      const each = arrayFunctions.each;
+      each([1], callBack);
+      expect(callBack).to.have.callCount(1);
+    });
   });
 
   describe('`map`', () => {
     it('should be a function', () => {
       const map = arrayFunctions.map;
-      expect(map).to.be.an('object');
+      expect(map).to.be.an('function');
+    });
+    it('should execute the callback at least once', () => {
+      const callBack = sinon.spy();
+      const map = arrayFunctions.map;
```

Comment:
Instead of declaring the `map` variable inside each `it()` statement, you can declare `map` once _outside_ the `it()` statement and reuse! (DRY — don't repeat your code!)

i.e.:
```
const { map } = arrayFunctions;

it('first', () => {
  expect(map)...
}
it('second', () => {
  expect(map)...
}
```